### ISSUE TITLE
Yarn fails install at building fresh packages error runnin optional dependency

### ISSUE THREAD

**Issue by bteepell** [yarnpkg/yarn#2174]
**bug**
Node 7.2.0 npm 4.0.2 yarn 0.17.10
Ubuntu 14 

Yarn install fails to finish installing .
at step [4/4] Building fresh packages I see err 
[1/1] ⠁ fsevents: gyp WARN install got an error, rolling back install
[-/1] ⠁ waiting...
[-/1] ⠁ waiting...
[-/1] ⠁ waiting...
error Error running install script for optional dependency after this constant loop of
$yarn install v0.17.10
[1/4] Resolving packages...
success Already up-to-date.
$ yarn install
yarn install v0.17.10
[1/4] Resolving packages...
success Already up-to-date.

I have tried removing node_modules and yarn.lock file and cleaning cache. 

I added optional fsevents to package.json
 "optionalDependencies": {
    "fsevents": "*"
  },

I tried yarn 0.18.0 as well. has same behavior but does remove the error for opt dependencies

**Comment by toddself** [comment]
Same issue here:

```
warning fsevents@1.0.15: The platform "linux" is incompatible with this module.
info "fsevents@1.0.15" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
[1/6] ⠠ libpq: CXX(target) Release/obj.target/addon/src/connection.o
[2/6] ⠠ fsevents: ERR!
```

It knows it shouldn't install it, but then it tries to build it anyway.

**Comment by kristojorg** [comment]
+1

**Comment by adiachenko** [comment]
Yeah, I get hard errors on Windows because of `fsevents` which shouldn't be installed anyway.

**Comment by intellix** [comment]
Also having this issue when deploying to Heroku. Everything installs fine on OSX and then pushing to Heroku it fails the build:

My yarn.lock contains this stuff:
```
chokidar@^1.4.1, chokidar@^1.4.3, chokidar@^1.6.0:
  version "1.6.1"
  resolved "https://registry.yarnpkg.com/chokidar/-/chokidar-1.6.1.tgz#2f4447ab5e96e50fb3d789fd90d4c72e0e4c70c2"
  dependencies:
    anymatch "^1.3.0"
    async-each "^1.0.0"
    glob-parent "^2.0.0"
    inherits "^2.0.1"
    is-binary-path "^1.0.0"
    is-glob "^2.0.0"
    path-is-absolute "^1.0.0"
    readdirp "^2.0.0"
  optionalDependencies:
    fsevents "^1.0.0"

fsevents@^1.0.0:
  version "1.0.15"
  resolved "https://registry.yarnpkg.com/fsevents/-/fsevents-1.0.15.tgz#fa63f590f3c2ad91275e4972a6cea545fb0aae44"
  dependencies:
    nan "^2.3.0"
    node-pre-gyp "^0.6.29"
```

**Comment by bteepell** [comment]
My issue I was having was related to #2335, the fsevents issue was fixed in latest release.   For me this issue is closed

**Comment by marcelogarbin** [comment]
I have a similar error...
![image](https://user-images.githubusercontent.com/9335588/39398112-5de799f8-4adf-11e8-8b29-b329cc4c3be1.png)

**Comment by lyminhooo** [comment]
Have same issuse! 

I check out with kibana
